esphome/components/toshiba/toshiba.cpp

```diff
@@ -439,6 +766,227 @@ void ToshibaClimate::transmit_rac_pt1411hwru_temp_(const bool cs_state, const bo
+        // Fan only mode supports fan speed control
+        message1[2] = (fan_code >> 8) & 0xFF;
+        message1[3] = fan_code & 0xFF;
+        message1[4] = RAS_2819T_FAN_ONLY_TEMP;
+        message1[5] = RAS_2819T_FAN_ONLY_TEMP_INV;
+        break;
+
+      default:
+        // Default case supports fan speed control
+        message1[2] = (fan_code >> 8) & 0xFF;
+        message1[3] = fan_code & 0xFF;
+        message1[4] = temp_code;
+        message1[5] = ~temp_code;
+        break;
+    }
+
+    // Build second packet (RAS_2819T_HEADER2 + 4 bytes)
+    message2[0] = RAS_2819T_HEADER2;
+
+    // Get fan speed encoding for rc_code_2
+    Ras2819tSecondPacketCodes second_packet_codes = get_ras_2819t_second_packet_codes(effective_fan_mode);
+    ESP_LOGD(TAG, "Second packet codes: fan_byte=0x%02X, suffix=0x%02X,0x%02X,0x%02X", second_packet_codes.fan_byte,
+             second_packet_codes.suffix.byte3, second_packet_codes.suffix.byte4, second_packet_codes.suffix.byte5);
+
+    // Determine header byte 2 and fan encoding based on mode
+    switch (this->mode) {
+      case climate::CLIMATE_MODE_COOL:
+        message2[1] = second_packet_codes.fan_byte;
+        message2[2] = 0x00;
+        message2[3] = second_packet_codes.suffix.byte3;
+        message2[4] = second_packet_codes.suffix.byte4;
+        message2[5] = second_packet_codes.suffix.byte5;
+        break;
+
+      case climate::CLIMATE_MODE_HEAT:
+        message2[1] = second_packet_codes.fan_byte;
+        message2[2] = 0x00;
+        message2[3] = second_packet_codes.suffix.byte3;
+        message2[4] = 0x00;
+        message2[5] = RAS_2819T_HEAT_SUFFIX;
+        break;
+
+      case climate::CLIMATE_MODE_HEAT_COOL:
+      case climate::CLIMATE_MODE_DRY:
+        // Auto/Dry modes use fixed values regardless of fan setting
+        message2[1] = RAS_2819T_AUTO_DRY_FAN_BYTE;
+        message2[2] = 0x00;
+        message2[3] = 0x00;
+        message2[4] = 0x00;
+        message2[5] = RAS_2819T_AUTO_DRY_SUFFIX;
+        break;
+
+      case climate::CLIMATE_MODE_FAN_ONLY:
+        message2[1] = second_packet_codes.fan_byte;
+        message2[2] = 0x00;
+        message2[3] = second_packet_codes.suffix.byte3;
+        message2[4] = 0x00;
+        message2[5] = RAS_2819T_HEAT_SUFFIX;
+        break;
+
+      default:
+        message2[1] = second_packet_codes.fan_byte;
+        message2[2] = 0x00;
+        message2[3] = second_packet_codes.suffix.byte3;
+        message2[4] = second_packet_codes.suffix.byte4;
+        message2[5] = second_packet_codes.suffix.byte5;
+        break;
+    }
+  }
+
+  // Log final messages being transmitted
+  ESP_LOGD(TAG, "[TX] Final message1: %02X %02X %02X %02X %02X %02X", message1[0], message1[1], message1[2],
+           message1[3], message1[4], message1[5]);
+  if (this->mode != climate::CLIMATE_MODE_OFF) {
+    ESP_LOGD(TAG, "[TX] Final message2: %02X %02X %02X %02X %02X %02X", message2[0], message2[1], message2[2],
```

Comment:
For messages like this, we usually use very_verbose:
```suggestion
  ESP_LOGVV(TAG, "[TX] Final message1: %02X %02X %02X %02X %02X %02X", message1[0], message1[1], message1[2],
           message1[3], message1[4], message1[5]);
  if (this->mode != climate::CLIMATE_MODE_OFF) {
    ESP_LOGVV(TAG, "[TX] Final message2: %02X %02X %02X %02X %02X %02X", message2[0], message2[1], message2[2],
```